src/classes/AFFL_AccRecordType_TDTM.cls

```diff
@@ -154,7 +154,9 @@ public class AFFL_AccRecordType_TDTM extends TDTM_Runnable {
         if(accsTypeAdded.size() > 0) {
             String childrenAccsTypeAddedQuery = 'select Account__c, Contact__c';
             for(Integer i = 0; i < accsTypeAdded.values().size(); i++) {
-                childrenAccsTypeAddedQuery += ', Contact__r.' + accsTypeAdded.values()[i];
+                if(!childrenAccsTypeAddedQuery.contains(accsTypeAdded.values()[i])) {
```

Comment:
This should work but I am not very comfortable with using string.contains(). Strings are likely unreliable and might cause issues.

Reply:
hmm, any suggestion here? Any specific concern?

Reply:
I don't know maybe a set of strings should work.
If you look at that method, we don't really use that map much. I am not sure if this is the right time to refactor this method but honestly I am not a big fan of it.

Reply:
Agree that we need to change the logic a little bit since I think I might found a logic error in this function. But I do think we need the map in this function since we want the Account Ids and the corresponding primary field. So, we need the map to connect Account Id and its primary field.  Let's schedule sometime to discuss more. 

1. We might not need to query the Contact fields since we do not use it in the following part.
2. We might not want to loop through every primary field in the values() for each contact. We need to use the map to find the corresponding primary field for the contact.